situp-plugins/elasticsearch/src/test/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSinkIT.java

```diff
@@ -276,4 +328,111 @@ private boolean deleteDirectory(final File directoryToBeDeleted) {
     }
     return directoryToBeDeleted.delete();
   }
+
+  public static boolean isODFE() {
+    final boolean isODFE = Optional.ofNullable(System.getProperty("odfe"))
+            .map("true"::equalsIgnoreCase).orElse(false);
+    if (isODFE) {
+      // currently only external cluster is supported for security enabled testing
+      if (!Optional.ofNullable(System.getProperty("tests.rest.cluster")).isPresent()) {
+        throw new RuntimeException("cluster url should be provided for security enabled ODFE testing");
+      }
+    }
+
+    return isODFE;
+  }
+
+  @Override
+  protected String getProtocol() {
+    return isODFE() ? "https" : "http";
+  }
+
+  @Override
+  protected RestClient buildClient(final Settings settings, final HttpHost[] hosts) throws IOException {
+    final RestClientBuilder builder = RestClient.builder(hosts);
+    if (isODFE()) {
+      configureHttpsClient(builder, settings);
+    } else {
+      configureClient(builder, settings);
```

Comment:
can we always call configure client and only configure the https stuff if needed

Reply:
I doublechecked the `configureClient` in `ESRestTestCase`. It turns out to contain keystore config stuff which we are really not using in ODFE integration tests. To keep settings more clear for ODFE. I tend to keep `configureHttpsClient` separate from the counterpart in the framework.